src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -772,10 +772,17 @@ internal static string GetRemotePrompt(RemoteRunspace runspace, string basePromp
             {
                 return basePrompt;
             }
-            else
+            else if (runspace.ConnectionInfo is SSHConnectionInfo)
             {
-                return string.Format(CultureInfo.InvariantCulture, "[{0}]: {1}", runspace.ConnectionInfo.ComputerName, basePrompt);
+                SSHConnectionInfo sshConnectionInfo = runspace.ConnectionInfo as SSHConnectionInfo;
+                if (!string.IsNullOrEmpty(sshConnectionInfo.UserName) &&
+                    !System.Environment.UserName.Equals(sshConnectionInfo.UserName, StringComparison.OrdinalIgnoreCase))
```

Comment:
For Windows sshConnectionInfo.UserName can have a domain component (domainName\userName), so we need to handle that unless we make this Linux platform only.  We can create a common helper method from here: https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs#L2052.  Or maybe update SSHConnectionInfo to automatically parse and expose UserName and DomainName properties.

Reply:
I am assuming we would want the prompt to look something like: `[user@domain@computer]: PS C:\>`

Reply:
Or maybe just drop the domain...

Reply:
I think you display it as `[<username>@<hostname>]` where `<username>` may or may not specify the additional `@domain`.  That's what I see from Bash.

Reply:
Currently, it depends on how the user specified the domain name.  If they used `user@domain`, it would be `[user@domain@computer]`.  Alternatively, if they used `domain\user`, it shows up as `[domain\user@computer]`.  Personally, I think we should show the domain and fine with whichever form the user provided.

Reply:
We don't currently support `user@domain` format.  It seems like domain information is not needed, but I am fine with `[domain\user@computer`].  We can easily change it later if it is unpopular.